rest-mvc/src/main/java/com/intuit/tank/rest/mvc/rest/services/agent/AgentServiceV2Impl.java

```diff
@@ -0,0 +1,296 @@
+import com.intuit.tank.storage.FileStorageFactory;
+import com.intuit.tank.vm.settings.AgentConfig;
+import com.intuit.tank.vm.settings.TankConfig;
+import com.intuit.tank.vm.agent.messages.Header;
+import com.intuit.tank.vm.agent.messages.Headers;
+import com.intuit.tank.vm.agent.messages.AgentTestStartData;
+import com.intuit.tank.vm.agent.messages.AgentData;
+import com.intuit.tank.perfManager.workLoads.JobManager;
+import com.intuit.tank.service.util.ServletInjector;
+import com.intuit.tank.service.impl.v1.cloud.JobController;
+import com.intuit.tank.service.impl.v1.cloud.CloudController;
+import com.intuit.tank.api.model.v1.cloud.CloudVmStatus;
+import com.intuit.tank.api.model.v1.agent.TankHttpClientDefinitionContainer;
+import com.intuit.tank.api.model.v1.agent.TankHttpClientDefinition;
+import com.intuit.tank.rest.mvc.rest.controllers.errors.GenericServiceCreateOrUpdateException;
+import com.intuit.tank.rest.mvc.rest.controllers.errors.GenericServiceResourceNotFoundException;
+
+import com.amazonaws.xray.AWSXRay;
+import com.amazonaws.xray.entities.Segment;
+import com.amazonaws.xray.entities.Subsegment;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.logging.log4j.Logger;
+import org.apache.logging.log4j.LogManager;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import javax.annotation.Nonnull;
+import javax.servlet.ServletContext;
+
+@Service
+public class AgentServiceV2Impl implements AgentServiceV2 {
+
+    @Autowired
+    private ServletContext servletContext;
+
+    private static final Logger LOGGER = LogManager.getLogger(AgentServiceV2Impl.class);
+    private static final String HARNESS_JAR = "apiharness-1.0-all.jar";
+    @Override
+    public String ping() {
+        return "PONG " + getClass().getInterfaces()[0].getSimpleName();
+    }
+
+    @Override
+    public String getSettings() {
+        String settings;
+        try {
+            File configFile = new TankConfig().getSourceConfigFile();
+            File agentConfigFile = new File(configFile.getParentFile(), "agent-settings.xml");
+            if (agentConfigFile.exists() && agentConfigFile.isFile()) {
+                configFile = agentConfigFile;
+            }
+            settings = FileUtils.readFileToString(configFile, StandardCharsets.UTF_8);
+        } catch (Exception e) {
+            LOGGER.error("Error reading agent settings file: " + e);
+            throw new GenericServiceResourceNotFoundException("agent", "settings file", e);
```

Comment:
@Zakaria-Kofiro What HTTP response are we throwing when we encounter these `GenericServiceResourceNotFoundException`?

Reply:
Ignore, I just saw your exception handlers